### PR TITLE
ScrollTo functionality is for making an object or a area visible.

### PR DESCRIPTION
**Pull request by gokaysatir**
If the coordinate is already visible, then don't scroll to that direction (x or y).

Issue:
* On Writer, it scrolls to view to (0, y) position to get to the end of the file.
* If there is a space on the left (zoomed-out), then the view is rendered non-centered.

Change-Id: I81f320ff29d1c5c84f1452e2f59935c78e929ad5

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] I have run `make prettier-write` and formatted the code.
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/src/layer/tile/CanvasTileLayer.js [hunk 1]
```diff
@@ -2955,7 +2955,10 @@ L.CanvasTileLayer = L.Layer.extend({
 		center = center.subtract(this._map.getSize().divideBy(2));
 		center.x = Math.round(center.x < 0 ? 0 : center.x);
 		center.y = Math.round(center.y < 0 ? 0 : center.y);
-		this._map.fire('scrollto', {x: center.x, y: center.y});
```
[the remark this answers is not in the document]
Comment (gokaysatir) [reply browser/src/layer/tile/CanvasTileLayer.js:2954]: Yep, corrected, thank you.

File: browser/src/layer/tile/CanvasTileLayer.js [hunk 2]
```diff
@@ -2948,10 +2948,33 @@ L.CanvasTileLayer = L.Layer.extend({
...
+		if (app.file.viewedRectangle.cY1 <= controlYUp && controlYDown <= app.file.viewedRectangle.cY2)
+			center.y = app.file.viewedRectangle.cY1;
+		else {
+			center.y -= this._map.getSize().divideBy(2).y;
+			center.y = Math.round(center.y < 0 ? 0 : center.y);
+		}
+
+		const section = app.sectionContainer.getSectionWithName(L.CSections.Scroll.name);
+		if (section) {
+			section.onScrollTo({x: center.x, y: center.y});
```
Comment (eszkadev) [inline browser/src/layer/tile/CanvasTileLayer.js:2976]: can we avoid to call it if both X and Y wasn't changed? as we already do these checks above

**Review by eszkadev — APPROVED** [review]
nice cleanup, seems to work ~ correctly, but I found corner cases with split panes:
[RecApp-2025-04-25-08:03:28.webm](https://github.com/user-attachments/assets/516b868e-20ac-40cd-b916-3c30778633f8)

(I used different zoom too)

**Comment by gokaysatir** [comment]
> nice cleanup, seems to work ~ correctly, but I found corner cases with split panes: [RecApp-2025-04-25-08:03:28.webm](https://github.com/user-attachments/assets/516b868e-20ac-40cd-b916-3c30778633f8)
> 
> (I used different zoom too)

Thanks for the catch :) Failing tests passed locally. I will retry, and i am checking this issue now.

File: browser/src/layer/tile/CanvasTileLayer.js (hunk 2 above)
Comment (gokaysatir) [reply browser/src/layer/tile/CanvasTileLayer.js:2976]: Yes we can improve that way. Let me also do this while i check the issue with multiple users.
